subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultArtifactCaches.java

```diff
@@ -82,7 +88,7 @@ private static File validateReadOnlyCache(DocumentationRegistry documentationReg
                 CacheLayout.ROOT.getKey() + " directory at " + root + " . Please follow the instructions at " + docLink);
             return null;
         }
-        return cacheDir;
+        return cacheDir.getAbsoluteFile();
```

Comment:
Why is this needed now?

Reply:
I want `cacheDir` to be an absolute path. Currently, it is read from whatever the environment variable is. That is why I think it is a good idea to convert it into an absolute path here.

Reply:
Why not do this where we call `validateReadOnlyCache()` from?

Reply:
Doing the `absoluteFile` now outside of `validateReadOnlyCache`.